proxy.py

```diff
@@ -1350,18 +1350,16 @@ def handle_readables(self, readables: List[socket.socket]) -> bool:
                                             'Upgraded client conn for plugin %s', str(plugin_))
                             elif isinstance(upgraded_sock, bool) and upgraded_sock:
                                 return True
+                except HttpProtocolException as e:
+                    logger.exception(
+                        'HttpProtocolException type raised', exc_info=e)
+                    response = e.response(self.request)     # type: ignore
```

Comment:
You can also remove `# type: ignore` now.  I had to add it because previously we were catching generic top-level `Exception`.  It shouldn't be required now as `HttpProtocolException` do have a `response` method.